controller/cache/cache.go

```diff
@@ -1767,20 +1767,24 @@ func startWorkerThread(ctx *Context) {
 									cache.errCount++
 									if cache.errCount >= scannerClearnupErrorMax {
 										log.WithFields(log.Fields{"scanner": sid}).Info("Remove stalled internal scanner")
-										_ = clusHelper.DeleteScanner(sid)
+										// Remove from cache first to avoid double cleanup when receiving our own delete notification
+										RemoveScannerFromCache(sid)
```

Comment:
In `func RemoveScannerFromCache(scannerID string)` it tries to grab cache lock for write by calling `cacheMutexLock()`.
But in line 1758 the cache lock is already owned by read.
Isn't this a deadlock?